STIPS/STIP-009.md

```diff
@@ -0,0 +1,188 @@
+- Allow owners to create manager smart contract and initialize and parameterize all modules and extensions in two transactions
+
+**BaseManagerV3**
+
+- Allow owners to permission specific functionality (extensions) to different operators
+- Allow owners to update permissions on specific functionality (extensions) for different operators
+- Allow owners to limit operator functionality using extensions with an asset whitelist
+- Allow owners to update asset whitelist
+
+**BasicIssuanceExtension**
+
+- Allow owners to enable functionality of BasicIssuanceModule with only a state change and no contract deployment
+
+**StreamingFeeExtension**
+
+- Allow owners to enable functionality of StreamingFeeModule with only a state change and no contract deployment
+
+**TradeExtension**
+
+- Allow owners to enable functionality of TradeModule with only a state change and no contract deployment
+
+## User Flows
+
+### ManagerFactory.create()
+
+An owner wants to create a new Set Token with a BaseManagerV3 smart contract manager.
+
+1. The `owner` calls create() passing in parameters to create a Set Token, parameters for the permissioning on BaseManagerV3, and the desired extensions. Specifically,
+
+    - components: List of addresses of components for initial positions
+    - units: List of units for initial positions
+    - name: Name of SetToken
+    - symbol: Symbol of SetToken
+    - owner: The address of the `owner`
+    - methodologist: The address of the `methodologist`
+    - operators: List of addresses of the `operator`s
+    - assets: List of addresses of assets for initial asset whitelist
+    - extensions: List of addresses of global extensions to be enabled
+
+2. A Set Token is deployed using SetTokenCreator
+3. A BaseManagerV3 is deployed
+4. Initialization parameters for the Set Token are stored on the Factory
+5. The Set Token and Manager are put in pending state
+
+### ManagerFactory.migrate()
+
+An owner wants to migrate an existing Set Token to a BaseManagerV3 smart contract manager.
+
+1. The `owner` calls migrate() passing in the Set Token address, parameters for the permissioning on BaseManagerV3, and the desired extensions. Specifically,
+
+    - owner: The address of the `owner`
+    - methodologist: The address of the `methodologist`
+    - operators: List of addresses of the `operator`s
+    - assets: List of addresses of assets for initial asset whitelist
+    - extensions: List of addresses of global extensions to be enabled
+
+2. A BaseManagerV3 is deployed
+3. Initialization parameters for the Set Token are stored on the Factory
+4. The Set Token and Manager are put in pending state
+
+### ManagerFactory.initialize()
+
+An owner wants to enable all extensions and initialize all corresponding modules.
+
+1. The `owner` calls initialize() passing in the parameters for initializing modules and extensions
+2. All modules are initialized
+3. All extensions are enabled
+4. The Set Token and Manager are put in initialized state
+
```

Comment:
One thing we need to do here is set the `owner` we won't be setting that in `create` or `migrate` since we need the factory to be able to initialize the modules via the extensions (which will only be doable by the `owner`).